### PR TITLE
Test doc cleanup

### LINKED ISSUE #334: No indication of skipped tests in main docs
'Running the tests' does not mention `xit` or `xdescribe`, and the installation steps currently gets you jasmine-node v1.14.3 which does not mark `xit` tests as skipped. The readme for the exercise does mention `xit` and `describe`, however someone who has already done the same exercise on another track (such as myself!) may go straight for the exercise and miss these details.

Test output from jasmine-node v1.14.3 suggests that you have completed an exercise even if it contains `xit` tests:
```
jasmine-node .
.

Finished in 0.005 seconds
1 test, 1 assertion, 0 failures, 0 skipped
```

jasmine-node v2 informs you that `xit` tests have been skipped:
```
jasmine-node .
.--

Finished in 0.004 seconds
3 Tests, 0 Failures, 2 Skipped
```

Should the installation instructions be updated to have `npm install jasmine-node@2 -g`? (I'm not very experienced with JavaScript so am unaware of any potential issues this could cause since it's beta). `docs/TESTS.md` should also be updated to mention `xit` and `xdescribe`.

### PR DESCRIPTION
**Pull request by tejasbubane**
Made some cleanup of the doc. Few of these cleanups are opinionated, so feel free to discuss.

* Removed the exercise fetching content since it does not belong here and is already present in the [Available exercises section](http://exercism.io/languages/javascript/exercises).
* Also removed the module exporting section coz we have stubs for a few beginning exercises. Kept the link to node module documentation for reference.
* Mentioned about skipped exercises and hence TDD. Closes #334 
* Mentioned that the other two sections - `Visual studio for windows` and `Linting` are optional.

### CODE REVIEW

**Comment by tejasbubane** [comment]
Ah, crap I forgot to `[ci-skip]` again... 😞

File: docs/TESTS.md [hunk 1]
```diff
@@ -1,37 +1,23 @@
...
 
     cd ~/exercism/javascript/bob
     jasmine bob_test.spec.js
 
-*Note that `~/exercism` is the default folder for `EXERCISM_HOME_DIR`. Be sure you use your configured folder for it*
-
-## Making Your First Node Module
-
-To create a module that can be loaded with `var Bob = require('./bob.js');`, put this code in `bob.js`:
+Only the first test will be executed, all the others have been marked pending using `xit` or `xdescibe`.
```
Comment (matthewmorgan) [inline docs/TESTS.md]: Should be `xdescribe`

File: docs/TESTS.md [hunk 2]
```diff
@@ -1,37 +1,23 @@
-After configuring `exercism` command-line client you can fetch the very next exercise
+WIth the first few exercises, you will get a skeleton file with the exported modules.
```
Comment (matthewmorgan) [inline docs/TESTS.md]: Small typo `WIth` -> `With`

**Comment by matthewmorgan** [comment]
@tejasbubane this looks like a good update to me!  I had two small callouts for typos that should be fixed before merging.

@rchavarria any further suggestions?

**Comment by tejasbubane** [comment]
Thanks @matthewmorgan. Fixed those typos.

**Comment by matthewmorgan** [comment]
@tejasbubane this is looking good,  I wanted to give @rchavarria a chance to comment if he has a chance, but I won't hold the PR too much longer if not.

File: docs/TESTS.md [hunk 3]
```diff
@@ -1,37 +1,23 @@
...
+For later ones, you can find more information about modules in the [node documentation](http://nodejs.org/api/modules.html#modules_module_exports).
 
-    exercism fetch javascript
-
-or you can fetch a specific exercise, passing the exercise name after the language. For example, to fetch the `bob` exercise:
-
-    exercism fetch javascript bob
-
-Now, it's time to run some tests. Move to the folder where that exercise's files are located (a path similar to `<EXERCISM_HOME_DIR>/<TRACK_ID>/<EXERCISE>`) and run the tests with the `jasmine` command you should have installed on the *Installing JavaScript* step:
+For running tests, you have installed `jasmine` from the [Installing Javascript section](http://exercism.io/languages/javascript/installing),
```
Comment (matthewmorgan) [inline docs/TESTS.md]: I just noticed this, which probably should be reworded a bit, maybe:

> For running tests, install `jasmine` as described in the [Installing Javascript section](...)

**Comment by tejasbubane** [comment]
@matthewmorgan I reworded that line and also squashed all commits together.

**Comment by rchavarria** [comment]
About the exercise fetching instructions, it was introduced to describe a clear way of working with exercises. I know that it's duplicated in other places, so I have no real issue of removing it. So, let's see how that documentation works for users.

👍 to skipped tests. Awesome!

About Visual Studio instructions, they were there before I started helping with this track. I'm not sure it they are still relevant or not. If nobody complains, I'm ok keeping them. 👍 to comment they are optional.

So, for me, the PR is ok. It seems documentation is as dynamic as code 😃

**Pull request merged**